Spurious error formatting bytes subclass with overridden __str__

* Please insert below the code you are checking with mypy.

```python
class HexBytes(bytes):
    def __str__(self) -> str:
        return self.hex()

value = HexBytes(b"aaa")
print(f"{value}")
```

* What is the actual behavior/output?

```
$ python3 mypy_bytes_subclass.py 
616161

$ mypy mypy_bytes_subclass.py 
mypy_bytes_subclass.py:6: error: On Python 3 '{}'.format(b'abc') produces "b'abc'"; use !r if this is a desired behavior
Found 1 error in 1 file (checked 1 source file)
```

* What is the behavior/output you expect?

I would expect no error to be produced in this situation, as `__str__` has been overridden, so the code does not exhibit the behaviour mentioned in the error.

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?

mypy 0.760, python 3.7.5
The same issue appears with `mypy-0.770+dev.e7866d0839202234db1f975a68a9284bb59d8544` (installed from git)

--- a/mypy/checkstrformat.py
+++ b/mypy/checkstrformat.py
@@ -382,7 +382,8 @@ class StringFormatterChecker:
                     self.msg.requires_int_or_char(call, format_call=True)
         if (not spec.type or spec.type == 's') and not spec.conversion:
             if self.chk.options.python_version >= (3, 0):
-                if has_type_component(actual_type, 'builtins.bytes'):
+                if (has_type_component(actual_type, 'builtins.bytes') and
+                        not custom_special_method(actual_type, '__str__')):
                     self.msg.fail("On Python 3 '{}'.format(b'abc') produces \"b'abc'\";"
                                   " use !r if this is a desired behavior", call,
                                   code=codes.STR_BYTES_PY3)

--- a/test-data/unit/check-expressions.test
+++ b/test-data/unit/check-expressions.test
@@ -1556,6 +1556,12 @@ def func(x: A) -> A:
 '{!r}'.format(b)
 '{!r}'.format(x)
 '{!r}'.format(n)
+
+class D(bytes):
+    def __str__(self) -> str:
+        return "overrides __str__ of bytes"
+
+'{}'.format(D())
 [builtins fixtures/primitives.pyi]
 
 [case testFormatCallFormatTypesBytesNotPy2]

